src/pool-service/IrcConnectionPool.ts

```diff
@@ -89,11 +88,12 @@ export class IrcConnectionPool {
                 };
             }
 
-            socket = await new Promise((resolve, reject) => {
+            return await new Promise((resolve, reject) => {
```

Comment:
Is it possible for line 122 to be reached without any of the `resolve`s having been hit?